src/marks/raster.d.ts

```diff
@@ -1,54 +1,258 @@
+  y: number[],
+
+  /** An array of values representing the sample’s observed value. */
+  values: any[]
 ) => any[];
 
+/** A spatial interpolation method. */
 export type RasterInterpolate = RasterInterpolateName | RasterInterpolateFunction;
 
+/**
+ * A source of pseudo-random numbers in [0, 1). The default source is seeded to
+ * ensure reproducibility.
+ */
 export type RandomSource = () => number;
 
+/**
+ * A sampler function, which returns a value for the given *x* and *y* values in
+ * the current *facet*. For example, to create side-by-side color plots of two
+ * functions of *x* and *y*:
+ *
+ * ```js
+ * Plot.raster({
+ *   fill: (x, y, {fx}) => fx(x) * y,
+ *   fx: [Math.sin, Math.cos],
+ *   x1: 0,
+ *   y1: 0,
+ *   x2: 4 * Math.PI,
+ *   y2: 2,
+ *   interval: 0.2
+ * });
+ * ```
+ */
 export type RasterSampler = (x: number, y: number, facet: number[] & {fx: any; fy: any}) => any;
 
+/** Options for the raster mark. */
 export interface RasterOptions extends Omit<MarkOptions, "fill" | "fillOpacity"> {
+  /** The horizontal coordinate of a sample. */
   x?: ChannelValueSpec;
+
+  /** The vertical coordinate of a sample. */
   y?: ChannelValueSpec;
+
+  /**
+   * The lower value of *x*, on the left edge of the raster (right edge if
+   * reversed). Defaults to the minimum of the **x** channel, if given,
+   * otherwise to 0, enabling to set up the raster coordinate system with a
+   * **width**.
+   */
   x1?: number;
+
+  /**
+   * The higher value of *x*, on the right edge of the raster (left edge if
+   * reversed). Defaults to the maximum of the **x** channel, if given,
+   * otherwise to **width**, enabling to set up the raster coordinate system
+   * with a **width**.
+   */
   x2?: number;
+
+  /**
+   * The lower value of *y*, on the top edge of the raster (bottom edge if
+   * reversed). Defaults to the minimum of the **y** channel, if given,
+   * otherwise to 0, enabling to set up the raster coordinate system with a
+   * **height**.
+   */
   y1?: number;
+
+  /**
+   * The higher value of *y*, on the bottom edge of the raster (top edge if
+   * reversed). Defaults to the maximum of the **y** channel, if given,
+   * otherwise to **height**, enabling to set up the raster coordinate system with a
+   * **height**.
+   */
   y2?: number;
+
+  /**
+   * The width of the raster grid, in actual columns of pixels. The grid might
+   * be scaled to the frame’s dimensions.
+   */
   width?: number;
+
+  /**
+   * The height of the raster grid, in actual rows of pixels. The grid might be
+   * scaled to the frame’s dimensions.
+   */
   height?: number;
+
+  /**
+   * The screen size of a raster pixel, used to determine the height and width
+   * of the raster from the frame’s dimensions; defaults to 1
+   */
   pixelSize?: number;
+
+  /**
+   * The sampling interval, used to determine the height and width of the raster
+   * from the frame’s dimensions and the *x* and *y* extents.
+   */
+  interval?: Interval;
```

Comment:
The raster mark doesn’t support the **interval** option; that’s only for the contour mark.